roles/ldapserver/tasks/main.yml

```diff
@@ -39,3 +39,197 @@
   loop:
     - people
     - groups
+
+- name: Place TLS cert
+  ansible.builtin.copy:
+    src: "{{ ldap_tls_cert_file }}"
+    group: openldap
+    mode: preserve
+    dest: /etc/ldap/ldaps.pem
+    remote_src: false
+  when: ldap_tls_cert_file
+
+- name: Place TLS private key
+  ansible.builtin.copy:
+    src: "{{ ldap_tls_key_file }}"
+    group: openldap
+    mode: 0640
+    dest: /etc/ldap/ldaps.key
+    remote_src: false
+  when: ldap_tls_key_file
+
+- name: Ensure cacert.pem
+  ansible.builtin.assemble:
+    src: /etc/ssl/certs/
+    regexp: "ICPC_WF_*"
+    group: openldap
+    mode: 0644
+    dest: /etc/ldap/cacert.pem
+  when:
+    - ldap_tls_cert_file
+    - ldap_tls_key_file
+
+- name: check for certinfo
+  ansible.builtin.shell:
+    cmd: ldapsearch -H ldapi:/// -LLL -Y EXTERNAL -b "cn=config" | grep -q /etc/ldap/ldaps.key
+  register: certinfo_present
+  changed_when: false
+  ignore_errors: true
+  when:
+    - ldap_tls_cert_file
+    - ldap_tls_key_file
+
+- name: Create certinfo
+  ansible.builtin.copy:
+    content: |
+      dn: cn=config
+      add: olcTLSCACertificateFile
+      olcTLSCACertificateFile: /etc/ldap/cacert.pem
+      -
+      add: olcTLSCertificateFile
+      olcTLSCertificateFile: /etc/ldap/ldaps.pem
+      -
+      add: olcTLSCertificateKeyFile
+      olcTLSCertificateKeyFile: /etc/ldap/ldaps.key
+    dest: /tmp/certinfo.ldif
+    mode: 0640
+  when:
+    - ldap_tls_cert_file
+    - ldap_tls_key_file
+    - certinfo_present.rc != 0
+
+- name: Add certinfo.ldif
+  ansible.builtin.command:
+    cmd: ldapmodify -Y EXTERNAL -H ldapi:/// -f /tmp/certinfo.ldif
+  when:
+    - ldap_tls_cert_file
+    - ldap_tls_key_file
+    - certinfo_present.rc != 0
+
+- name: Ensure TLS in slapd
+  ansible.builtin.lineinfile:
+    path: /etc/default/slapd
+    regexp: '^SLAPD_SERVICES=.*'
+    line: 'SLAPD_SERVICES="ldap:/// ldapi:/// ldaps:///"'
+    owner: root
+    group: root
+    mode: '0644'
+  when:
+    - ldap_tls_cert_file
+    - ldap_tls_key_file
+  notify: restart slapd
+
+- name: Ensure ldap dirs are present
+  ansible.builtin.file:
+    path: "/var/lib/ldap/accesslog"
+    state: directory
+    owner: openldap
+    mode: 0750
+  when: ldap_provider
+
+- name: provider replicator setup
+  community.general.ldap_entry:
+    dn: cn=replicator,{{ ldap_domain_dc }}
+    objectClass:
+      - simpleSecurityObject
+      - organizationalRole
+    attributes:
+      description: Replication user
+      userPassword: "{CRYPT}x"
+    bind_dn: cn=admin,{{ ldap_domain_dc }}
+    bind_pw: "{{ ldap_admin_password }}"
+
+- name: set password for replicator
+  community.general.ldap_passwd:
+    bind_dn: cn=admin,{{ ldap_domain_dc }}
+    bind_pw: "{{ ldap_admin_password }}"
+    dn: cn=replicator,{{ ldap_domain_dc }}
+    passwd: "{{ ldap_replicator_password }}"
+  when: ldap_replicator_password
+
+- name: Set up the ACL
```

Comment:
All these use the same when, so consider using a block with that when for better readability + a small speedup

Reply:
moved to use blocks